### DISCUSSION TITLE
How to manage multiple domains with NextAuth.js?

### DISCUSSION THREAD

**Discussion by jacobhjkim** [nextauthjs/next-auth#4022] [Help] [answered]
### Question 💬

I have two Next.js apps `example.com` and `app.example.com`. I want to handle all the login logic in `example.com` and only allow authenticated users to view `app.example.com`. Can I achieve this with NextAuth.js?

Here's what I have came up with. Created a Next.js middleware in `app.example.com` and also create `app.example.com/api/auth`. However, this will allow users to login in the `app.example.com` and it seems weird. Is there a better way to manage this?

**Comment by balazsorban44** [comment, marked as answer]
Fully possible. Best answer right now: https://github.com/nextauthjs/next-auth/discussions/1299#discussioncomment-362054

We would like to make this easier in the future.

**Reply by jacobhjkim** [reply]
Awesome! Thanks, I will check it out.

**Reply by lveillard** [reply]
Hello! is this still in the roadmap? Monorepos are getting more and more frequent. 

In my case I have two next apps that require next auth for their own pages and their own apis.  (Lets say docs.domain.com and app.domain.com) 

These two APIs then are calling a third next js app that is API only (api.domain.com). Ideally I would love to automatically authenticate their requests from docs, app & also external API calls with api-keys

I guess there is still no easy alternative to do this ? 🤔

I was thinking on using next-auth only in the 3rd app (api.domain.com) and then redirect (next async rewrittes in next.config.js) every /auth request from `docs` and `app` to  `api`. 

But not sure if I would be able to protect pages locally in each app 🤔

**Reply by rahul-269** [reply]
@lveillard How did you manage to do it in the end ? I am trying to do something similar, I also have two next apps that require seperate next auth for their own pages and apis. I tried to setup the auth , but its giving me several issues, the main one being if i have them running concurrently , if i login to one , then login to the other it destroys my session on the first one and logs me out.

**Reply by lveillard** [reply]
I had to split them in two different next auth instances :(

**Comment by vignesh-gupta** [comment]
@balazsorban44 quick question on this point: 

Context:
I've `app.clikz.live` as my main app and `clikz.live` for landing page. But once authenticated with app domain, the data is not available on base domain. I've setup `cookie ` option but still doesn't work.

Code - https://github.com/vignesh-gupta/clikz/blob/main/apps/web/auth.ts#L16
Live - https://clikz.live/

**Reply by vignesh-gupta** [reply]
My bad it work, I just hadn't did it correctly. Please ignore
